ssl/statem/extensions.c

```diff
@@ -1346,3 +1356,29 @@ static int final_early_data(SSL *s, unsigned int context, int sent, int *al)
 
     return 1;
 }
+
+static int final_maxfragmentlen(SSL *ssl, unsigned int context, int sent, int *al)
+{
+    if (!ssl->server)
+        return 1;
+
+    /* Session resumption on server-side with MFL extension active
+       BUT MFL extension packet not resent  */
+    if (ssl->hit && USE_MAX_FRAGMENT_LENGTH_EXT(ssl->session)
+            && !sent ) {
+        *al = SSL_AD_INTERNAL_ERROR;
+        return 0;
+    }
+
+    /* Current SSL buffer is lower than requested MFL */
+    if (sent
+            && ssl->max_send_fragment < GET_MAX_FRAGMENT_LENGTH(ssl->session)) {
+        size_t pipes = ssl->rlayer.numwpipes;
+
+        if (!ssl3_release_write_buffer(ssl))
+            return 0;
+        if (!ssl3_setup_write_buffer(ssl, pipes, 0))
```

Comment:
Why do we only update the write buffer here and not the read buffer? Thinking about it updating the read buffer at this point could be problematic if anything tries to use the read buffer for this message after extension processing.